### PR TITLE
Attributes need getter and setter

### PR DESCRIPTION
**Pull request by byeolbit**
Implement getter and setter for attributes.

Issue=#146

### CODE REVIEW

**Review by romandev — COMMENTED** [review]
Hmm.. I'm not sure but you seem to misunderstand my intention.

Let's imagine someone defines WebIDL as follows:
```webidl
interface Hello {
    readonly attribute long readonlyAttribute;
    attribute long readWriteAttribute;
};
```

Then, we can write C++ impl code as follows:
```c++
class Hello {
  public:
    long GetReadonlyAttribute() {
        // This is getter for readonlyAttribute
    }

    long GetReadWriteAttribute() {
        // This is getter for readWriteAttribute
    }

    void SetReadWriteAttribute() {
        // This is setter for readWriteAttribute
    }

    // I don't think you should add "public" member variables.
};
```

**Review by yjaeseok — APPROVED** [review]
lgtm

**Comment by byeolbit** [comment]
@romandev Then variables should be moved to private area?

File: test/test_interface.idl [hunk 1]
```diff
@@ -40,9 +40,14 @@ interface TestInterface {
   attribute double doubleNumber;
   readonly attribute double readonlyDoubleNumber;
   static attribute double staticDoubleNumber;
-  double getDoubleAttribute();
   void readonlyAssignTest(double number);
   static double StaticTest(double number);
+  double getDoubleNumber();
```
Comment (romandev) [inline test/test_interface.idl]: You already defined ```attribute double doubleNumber;``` but you also define getDoubleNumber() here.
This should be conflict. The ```attribute``` in WebIDL SHOULD be represented as Get/Set methods in C++ layer.

**Review by romandev — COMMENTED** [review]
@byeolbit The member variables should be ```private```. (Please see my inline comment :))

File: test/test_interface.h [hunk 2]
```diff
@@ -41,19 +41,23 @@ class TestInterface {
...
+  static double GetStaticDoubleNumber();
+  static void SetStaticDoubleNumber(double number);
 
  private:
   // FIXME(zino): Currently, we should set this variable in each methods. It's
   // not elegance way. We should find a way to get function name and signature
   // automatically. (I tried __FUNCTION__ and __PRETTY_FUNCTION__ but they are
   // dependent on each platform.
   static std::string last_call_info_;
+  double doubleNumber;
```
Comment (romandev) [inline test/test_interface.h]: We follows Chromium C++ style(including Google C++ style):
https://google.github.io/styleguide/cppguide.html#Variable_Names (Class Data Members)

So, could you please append a trailing underscore?

File: test/test_interface.cc [hunk 3]
```diff
@@ -70,14 +70,28 @@ void TestInterface::VoidMethodTestEnumArg(const std::string& string) {
...
 double TestInterface::StaticTest(double number) {
   return TestInterface::staticDoubleNumber;
 }
 
 void TestInterface::ReadonlyAssignTest(double number) {
   this->readonlyDoubleNumber = number;
+}
+
+double TestInterface::GetDoubleNumber() {
+  return this->doubleNumber;
```
Comment (romandev) [inline test/test_interface.cc]: We prefer to omit ```this->``` if possible.

**Review by romandev — COMMENTED** [review]
looks good to me overall

File: test/test_interface.h (hunk 2 above)
Comment (byeolbit) [reply test/test_interface.h]: I forgot to change it. I'll fix it.

Comment (romandev) [reply test/test_interface.h]: @byeolbit, Hmm, you had to append a trailing underscore '_' :)

File: test/test_interface.idl [hunk 4]
```diff
@@ -37,10 +37,9 @@ interface TestInterface {
   void voidMethodTestEnumArg(TestEnum enumValue);
 
   // Attributes
-  attribute double doubleNumber;
-  readonly attribute double readonlyDoubleNumber;
-  static attribute double staticDoubleNumber;
-  double getDoubleAttribute();
+  attribute double double_number;
```
Comment (romandev) [inline test/test_interface.idl]: In WebIDL, should use camel case instead of snake case!

File: template/interface_cpp.njk [hunk 5]
```diff
@@ -73,28 +73,28 @@ void {{name}}Bridge::Init(Napi::Env env, Napi::Object exports) {
...
 {% if member.member_type == "attribute" %}
-Napi::Value {{name}}Bridge::Get{{member.name | camelcase}}(const Napi::CallbackInfo& info) {
+Napi::Value {{name}}Bridge::Getter{{member.name | camelcase}}(const Napi::CallbackInfo& info) {
   auto return_value = 
   {% if member.is_static %}
-  {{name}}::{{member.name}};
+  {{name}}::Get{{member.name | camelcase}}();
   {% else %}
-  impl_-> {{member.name}};
+  impl_-> Get{{member.name | camelcase}}();
```
Comment (romandev) [inline template/interface_cpp.njk]: impl_->Get
(should remove unnecessary blank)

**Comment by byeolbit** [comment]
@romandev I changed code as your comments. PTAL :)

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**